test/runtests.jl

```diff
@@ -1,43 +1,40 @@
-using InfrastructureModels
-using PowerModels
-using GasModels
 using GasPowerModels
-using Memento
 
-# Suppress warnings during testing.
-setlevel!(getlogger(GasModels), "error")
+import Memento
+import MathOptInterface
+import InfrastructureModels
+import GasModels
+import PowerModels
 
-GasPowerModels.silence()
+const _MOI = MathOptInterface
```

Comment:
instead of this you could consider,
```
const _MOI = GasPowerModels._MOI
const _GM = GasPowerModels._GM
```
But it does beg the question, why do we need access to these packages in the tests?

It might be nice of basic GasPowerModels usage only required `using GasPowerModels` without knowing about the other dependent requirements.

Reply:
Good idea on grabbing the references from `GasPowerModels`. I think the reason we need to reference `_GM` and `_PM` in the tests is in the construction of the coupled model, which requires specifying the individual formulation types, e.g., `_GM.MISOCPGasModel` and `_PM.SOCWRPowerModel`. I'm open to removing these if there's a way we could be embedding the individual formulation types within `GasPowerModels`, though.

Reply:
Similar to what we did with MOI status codes, I would consider re-exporting these types from `GasPowerModels`, so that a user can provide `SOCWRPowerModel` or `MISOCPGasModel` to a `run_*` call without needing to load `PowerModels` and `GasModels`.

This approach worked for status codes, https://github.com/lanl-ansi/PowerModels.jl/blob/master/src/core/export.jl#L43 it might be tricker to automate for the model types.

Reply:
All right... tell me how this can be improved. I'd like to make this into a single function that reads in `GasModels` or `PowerModels` as a parameter instead of duplicating it twice, but I can't find a way to do it.
```julia
# Export final GasModels model types.
function _recursive_export_gas_models(model_type)
    subtypes = InteractiveUtils.subtypes(GasModels, model_type)

    if length(subtypes) > 0
        for model_type in subtypes
            _recursive_export_gas_models(model_type)
        end
    else
        model_type_str = replace("$(model_type)", "GasModels." => "")
        @eval import GasModels: $(Symbol(model_type_str))
        @eval export $(Symbol(model_type_str))
    end
end

_recursive_export_gas_models(GasModels.AbstractGasModel)

# Export final PowerModels model types.
function _recursive_export_power_models(model_type)
    subtypes = InteractiveUtils.subtypes(PowerModels, model_type)

    if length(subtypes) > 0
        for model_type in subtypes
            _recursive_export_power_models(model_type)
        end
    else
        model_type_str = replace("$(model_type)", "PowerModels." => "")
        @eval import PowerModels: $(Symbol(model_type_str))
        @eval export $(Symbol(model_type_str))
    end
end

_recursive_export_power_models(PowerModels.AbstractPowerModel)
```

Reply:
In terms of making this one function, did you try adding a second argument where you pass the module id?  If you leave it untyped I think it will work.

As for how this is implemented, instead of `InteractiveUtils`, what do you think about using some more basic tools like `isa` and `<:`.  For example, you could looper over every symbol in PowerModels and check if it is a type and if so, a subtype of `AbstractPowerModel`.

Reply:
Per our discussion, I'll loop over all types in `GasModels` and `PowerModels` and export anything that `isa <: AbstractGasModel` or `isa <: AbstractPowerModel`, respectively.

Reply:
Because everything returned with `names` is of type `Symbol`, using the `isa` method is a bit ugly: you have to import the symbol, check if it's of the correct type, then conditionally export. I like the below better, which accomplishes the same thing, although it uses string comparisons:
```julia
# Export GasModels modeling types for ease of use.
gas_models = filter(x -> endswith(string(x), "GasModel") &&
    !occursin("Abstract", string(x)), names(GasModels))
for x in gas_models @eval import GasModels: $(x) end
for x in gas_models @eval export $(x) end

# Export PowerModels modeling types for ease of use.
power_models = filter(x -> endswith(string(x), "PowerModel") &&
    !occursin("Abstract", string(x)), names(PowerModels))
for x in power_models @eval import PowerModels: $(x) end
for x in power_models @eval export $(x) end
```